source/geod24/RingBuffer.d

```diff
@@ -0,0 +1,127 @@
+module geod24.RingBuffer;
+
+import std.range;
+
+/*******************************************************************************
+
+    A ring buffer implementation with fixed size
+
+    Params:
+        T = Type of elements
+
+*******************************************************************************/
+
+class RingBuffer (T)
+{
+
+    /*******************************************************************************
+
+        Params:
+            size = Maximum number of elements
+
+    *******************************************************************************/
+
+    this (size_t size) @safe pure nothrow
+    {
+        this.storage = new T[size];
+        this.storage.length = size;
+        if (size)
```

Comment:
Why are we supporting ring buffers with zero capacity if using it will result in an error (exceptions thrown)?

Wouldn't it be better to just assert that size != 0?

Reply:
Just so that Channel can get away with not checking if it is operating in buffered or unbuffered mode. I will make Channel explicitly check its op mode and assert size != here.